File: 2211_03/dfs와bfs.py
# https://www.acmicpc.net/problem/1260
import sys, collections
def dfs(n):
    global dfs_seq
    for i in sorted(EDGES[n]):
        if not visited[i]:
            visited[i] = True
            dfs_seq.append(str(i))
            dfs(i)
def bfs(n):
    global bfs_seq
    q = collections.deque()
    q.append(n)
    visited = [False] * (N + 1)
    visited[n] = True
    while q:
        edge = q.popleft()
        for i in sorted(EDGES[edge]):
            if not visited[i]:
                visited[i] = True
                bfs_seq.append(str(i))
                q.append(i)

N, M, V = map(int, sys.stdin.readline().split())
EDGES = collections.defaultdict(list)
for _ in range(M):
    n1, n2 = map(int, sys.stdin.readline().split())
    EDGES[n1].append(n2)
    EDGES[n2].append(n1)

dfs_seq, bfs_seq = [str(V)], [str(V)]
visited = [False] * (N + 1)
visited[V] = True
dfs(V)
bfs(V)
print(' '.join(dfs_seq))
print(' '.join(bfs_seq))

File: 2211_03/dfs와bfs.java
// https://www.acmicpc.net/problem/1260
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
    static int N, M, V;
    static int[][] EDGES;
    static int[] visited;

    public static void main(String args[]) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        N = Integer.parseInt(st.nextToken());
        M = Integer.parseInt(st.nextToken());
        V = Integer.parseInt(st.nextToken());
        EDGES = new int[N + 1][N + 1];
        for(int i=0;i<M;i++){
            st = new StringTokenizer(br.readLine());
            int node1 = Integer.parseInt(st.nextToken());
            int node2 = Integer.parseInt(st.nextToken());
            EDGES[node1][node2] = EDGES[node2][node1] = 1;
        }

        // dfs
        visited = new int[N + 1];
        visited[V] = 1;
        System.out.print(V);
        dfs(V);
        System.out.println();

        // bfs
        Queue<Integer> q = new LinkedList<>();
        q.add(V);
        visited = new int[N + 1];
        visited[V] = 1;
        System.out.print(V);
        while (!q.isEmpty()){
            int node = q.poll();
            for(int i=1;i<N+1;i++){
                if(EDGES[node][i] == 1 && visited[i] == 0){
                    visited[i] = 1;
                    System.out.print(" " + Integer.toString(i));
                    q.add(i);
                }
            }
        }
    }
    public static void dfs(int node){
        for(int i=1;i<N+1;i++){
            if(EDGES[node][i] == 1 && visited[i] == 0){
                visited[i] = 1;
                System.out.print(" " + Integer.toString(i));
                dfs(i);
            }
        }
    }
}
